redis.go

```diff
@@ -285,7 +285,8 @@ func (c *baseClient) initConn(ctx context.Context, cn *pool.Conn) error {
 	// we continue to provide services with RESP2.
 	if err := conn.Hello(ctx, 3, username, password, "").Err(); err == nil {
 		auth = true
-	} else if !strings.HasPrefix(err.Error(), "ERR unknown command") {
+	} else if !strings.HasPrefix(err.Error(), "ERR unknown command") &&
+		!strings.HasPrefix(err.Error(), "NOAUTH Authentication") {
```

Comment:
@xinzhanguo could you a comment mentioning that this is check is for DragonflyDB?

Reply:
i checked, it's ok, DragonflyDB code is here https://github.com/dragonflydb/dragonfly/blob/8cf81151166865e2201fe10bf660e140e0e2fc13/src/server/main_service.cc#L612

Of course, the best way is that the third-party system supports `Hello` command or return standard error messages, but the reality is that the already running service will not be updated, even the code is modified, there will be problems when using v9 requests.

Reply:
Can code comments be added? This line of code is marked for compatibility with dragonfly.